core/src/main/java/org/carbondata/core/carbon/datastore/block/TableBlockInfo.java

```diff
@@ -185,6 +207,16 @@ public long getBlockLength() {
         > ((TableBlockInfo) other).blockOffset + ((TableBlockInfo) other).blockLength) {
       return 1;
     }
+    //compare the startBlockLetNumber
+    int diffStartBlockLetNumber =
+        blockletInfos.getStartBlockletNumber() - ((TableBlockInfo) other).blockletInfos
```

Comment:
@PraveenAdlakha  your point is valid. We can refactor this code. Can you work on it and provide the patch.